orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -102,19 +103,41 @@ public TaskResult execute(@Nonnull Stage stage) {
     StageData stageData = stage.mapTo(StageData.class);
     Map<String, Object> cluster;
 
+    AtomicReference<Moniker> moniker = new AtomicReference<>(stageData.moniker);
+    if (moniker.get() == null && stageData.serverGroup != null) {
+      try {
```

Comment:
handle rolling back to a server group that might not be cached yet (maybe the deploy failed prior to force cache refresh completing)